frontend/src/store/Books/BooksReducer.ts

```diff
@@ -21,18 +21,19 @@ const initialState = {
 const booksReducer = (
 	state: BooksState = initialState,
 	action: DispatchBookActions
-) => {
+):BooksState => {
 	switch (action.type) {
 		case BOOKS_LOADING: {
 			return {
+				...state,
 				loading: true,
 			};
 		}
 		case BOOKS_SUCCESS: {
 			return {
 				...state,
 				loading: false,
-				books: action.payload,
+				books: [action.payload],
```

Comment:
line 28, preserve the previous state. needed for output return
line 36, needed for the bookstate - book